test/stream.test.js

```diff
@@ -273,6 +273,48 @@ describe('StreamAPI', () => {
         })
         .start()
     })
+
+    test('stays open beyond the value set for http2SessionIdleTime', async () => {
+      let seen_event = false
+      const idleTime = 1000
+      const client = util.getClient({
+        secret: key.secret,
+        http2SessionIdleTime: idleTime,
+      })
+      const oldTimestamp = doc.ts
+
+      const assertActiveSessions = length =>
+        expect(Object.keys(client._http._adapter._sessionMap).length).toBe(
+          length
+        )
+
+      stream = client.stream
+      .document(doc.ref)
+      .on('version', (event) => {
+        seen_event = true
+        expect(event.action).toEqual("update")
+        expect(event.document.ts).toBeGreaterThan(oldTimestamp)
+      })
+
+      stream.start()
+
+      await util.delay(idleTime + 500)
+      assertActiveSessions(1)
+
+      const { ts: newTimestamp } = await client.query(q.Update(doc.ref, {}))
+      expect(newTimestamp).toBeGreaterThan(oldTimestamp)
+      await util.delay(idleTime + 1)
```

Comment:
Wouldn't it be worth checking that the stream is still open beyond the maximum value of `http2SessionIdleTime`? As written, this test doesn't really demonstrate that a stream is held open indefinitely, just that it is held open for 2002 ms (which is well within 5000ms).

Reply:
The test should be updated now